.travis.yml

```diff
@@ -21,20 +21,17 @@ env:
     - TEST_BLACKLIST="moveit_ros_perception"
     - CXXFLAGS="-Wall -Wextra -Wwrite-strings -Wunreachable-code -Wpointer-arith -Wredundant-decls"
     - WARNINGS_OK=false
-  jobs:
-    - TEST="clang-format catkin_lint"
-    - TEST=code-coverage
-    - ROS_DISTRO=melodic
-    - ROS_DISTRO=kinetic
 
-jobs: # Add a separate config to the matrix, using clang as compiler
+jobs:
   include:
-    # add a config to the matrix using clang as compiler and also test ikfast plugin creation
+    - env: TEST="clang-format catkin_lint"
+    - env: TEST=code-coverage
+    - env: ROS_DISTRO=melodic BEFORE_DOCKER_SCRIPT="source moveit_kinematics/test/test_ikfast_plugins.sh"
+    - env: ROS_DISTRO=kinetic
     - compiler: clang
-      env: ROS_REPO=ros-shadow-fixed
-           TEST=clang-tidy-fix
-           BEFORE_DOCKER_SCRIPT="source moveit_kinematics/test/test_ikfast_plugins.sh"
-           CXXFLAGS="-Wall -Wextra -Wwrite-strings -Wunreachable-code -Wpointer-arith -Wredundant-decls -Wno-overloaded-virtual"
+      env: TEST=clang-tidy-fix CXXFLAGS="-Wall -Wextra -Wwrite-strings -Wunreachable-code -Wpointer-arith -Wredundant-decls -Wno-overloaded-virtual"
+  allow_failures:
+    - env: ROS_REPO=ros-shadow-fixed
```

Comment:
You need to list this job in `include` as well. `allow_failures` is intended to mark an existing job - weird Travis logics...